src/lib/ReduxChannel.js

```diff
@@ -0,0 +1,86 @@
+const DEFAULT_OPTIONS = {
+  timeout: 5000,
+  target: window
+};
+
+class Channel {
+  constructor(options = {}) {
+    if (!options.incoming) throw new Error("You must specify an incoming event name with the 'incoming' option.");
+    if (!options.outgoing) throw new Error("You must specify an outgoing event name with the 'outgoing' option.");
+    this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.callbacks = new Set();
+    this.timeouts = new Map();
+
+    this._onReceiveEvent = this.runCallbacks.bind(this);
+    this.options.target.addEventListener(this.options.incoming, this._onReceiveEvent);
+  }
+
+  on(cb) {
+    this.callbacks.add(cb);
+  }
+
+  off(cb) {
+    if (cb) {
+      this.callbacks.delete(cb);
+    } else {
+      this.callbacks.clear();
+    }
+  }
+
+  runCallbacks(event) {
+    const action = event.detail;
+    this.callbacks.forEach(cb => cb(action));
+  }
+
+  broadcast(action) {
+    const event = new CustomEvent(this.options.outgoing, {detail: JSON.stringify(action)});
+    this.options.target.dispatchEvent(event);
+  }
+
+  connectStore(store) {
+    this.on(action => store.dispatch(action));
+  }
+
+  get middleware() {
+    return store => next => function (action) {
```

Comment:
currying: returns a function, which returns a function, but the last "function" is bound to the outer "this".